### PR TITLE
harpwise 4.4.4 (new formula)

### PR DESCRIPTION
**Pull request by marcIhm**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/h/harpwise.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+require "fileutils"
+
+class Harpwise < Formula
+  desc "Harmonica practice-utility for the command-line using microphone and speaker"
+  homepage "https://github.com/marcIhm/harpwise"
+  url "https://github.com/marcihm/harpwise/archive/4.4.4.tar.gz"
+  sha256 "7caae0c4732bb188d164d86a4cceb2aa7b4f09859b945bc86f42d31500571e83"
+  license "MIT"
+
+  # dependencies ordered according to "brew style harpwise"
```
Comment (chenrui333) [inline Formula/h/harpwise.rb:10]: this can be removed 

```suggestion
```

File: Formula/h/harpwise.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require "fileutils"
+
```
Comment (chenrui333) [inline Formula/h/harpwise.rb:2]:
```suggestion
```

File: Formula/h/harpwise.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  homepage "https://github.com/marcIhm/harpwise"
+  url "https://github.com/marcihm/harpwise/archive/4.4.4.tar.gz"
+  sha256 "7caae0c4732bb188d164d86a4cceb2aa7b4f09859b945bc86f42d31500571e83"
+  license "MIT"
+
+  # dependencies ordered according to "brew style harpwise"
+  depends_on "aubio"
+  depends_on "coreutils"
+  depends_on "figlet"
+  depends_on "ruby"
```
Comment (chenrui333) [inline Formula/h/harpwise.rb:14]: does it work with old ruby?

```suggestion
  uses_from_macos "ruby"
```

File: Formula/h/harpwise.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  depends_on "sox"
+  depends_on "toilet"
+
+  def install
+    # harpwise is written in ruby, but is not a gem.
+    man1.install "man/harpwise.1" => "harpwise.1"
```
Comment (chenrui333) [inline Formula/h/harpwise.rb:20]:
```suggestion
    man1.install "man/harpwise.1"
```

File: Formula/h/harpwise.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    mkdir "bin"
```
Comment (chenrui333) [inline Formula/h/harpwise.rb:21]:
```suggestion
```

Comment (marcIhm) [reply Formula/h/harpwise.rb:18]: This mkdir happens within the downloaded and unpacked archive; it is needed for the following File.write to work.

File: Formula/h/harpwise.rb (hunk 3 above)
Comment (marcIhm) [reply Formula/h/harpwise.rb:11]: Probably not with ruby 2.6; added a new commit to revert to ruby from homebrew. Thanx for the hint !

**Comment by marcIhm** [comment]
Hi chenrui333, thanx for your suggestions ! I could follow them all.

File: Formula/h/harpwise.rb (hunk 5 above)
Comment (marcIhm) [reply Formula/h/harpwise.rb]: 'mkdir "bin"' is gone, as I now create the stub within libexec.

**Comment by marcIhm** [comment]
Hi chenrui333, thanx for your suggestions ! I could follow them all.

File: Formula/h/harpwise.rb (hunk 3 above)
Comment (chenrui333) [reply Formula/h/harpwise.rb:11]: yeah, for os with old ruby, you can do special syntax `uses_from_macos "ruby", since: :monterey`

**Comment by marcIhm** [comment]
Hi Rui,
is there anything I can do about this ? Tests are failing because of low notability tag, about which I cannot do much ...
Of course, if you want to refelct or talk with others from the project, then just take your time.
Regards
Marc

**Comment by chenrui333** [comment]
We are definitely happy to accept the PR once it pass the notability check.

**Comment by marcIhm** [comment]
Hi Rui,
okay, I understand; no problem ! Probably I will go for a personal tap; maybe this will increase notability; and if not, then my software is truely niche :- )
In any case, thanx for your help in improving this formula !
regards
Marc

**Pull request closed without merging**